SQL in R - cheatsheet.R

```diff
@@ -0,0 +1,711 @@
+# 
+# Common problems/errors while writing SQL code in R:
+#   Problems with simple and double quotation marks. 
+#   Commas before FROM, or lack of commas between fields.
+#   No grouping by expressions or missing some, if you use any command to aggregate
+#   the data you will need a group by
+
+# Use the table of contents to get to the right bit of information
+
+# This script cover examples using native SQL within R, but another approach
+# is to use the package dbplyr which translates R code (based on dplyr) into SQL.
+
+###############################################.
+## Let's start ----
+###############################################.
+# You will need access to SMR01 and the deaths catalogue
+# To connect to a database first you need the package odbc
+library(odbc)
+library(dplyr) # and dplyr for a couple of things
+
+# And now connect to the database with your login details
+channel <- suppressWarnings(dbConnect(odbc(),  dsn="SMRA",
+                                      uid=.rs.askForPassword("SMRA Username:"), 
+                                      pwd=.rs.askForPassword("SMRA Password:")))
+
+# Show all tables/views available 
+dbListTables(channel)
+# Look at the table 'scheme' types that are available by running the code below
+odbcListObjects(channel)  
+# "Analysis" schemes are the analysis views, likely the ones you want to use. 
+# Tables with a 'scheme' which is a username are data uploaded by individual users. 
+# Show tables/views again, but this time restrict to "Analysis" views only.
+odbcListObjects(channel, schema="ANALYSIS")
+# You can now see the names of the analysis views, these are the same names/data 
+# Most of the views are versions of the various SMR databases: 
+# SMR00 (outpatients), SMR01 (Acute hospital activity), SMR04 etc.
+# The "_PI" suffix means that these views contain personally identifying 
+# information (names, dates of birth etc). SMRXX_PI is the usual version used 
+# for analyses. SMRA also includes NRS death records (GRO_DEATHS_C). 
+# Other files are reference files which contain information about coded fields
+# e.g. if you have a list of codes referring to hospitals and you want to know 
+# the hospital names or postcodes, you would use the LOCATION view to look up 
+# this information.
+# 
+# In order to view the variables, we can preview the table. 
+# Further details of variable names and attributes in each of the SMR views 
+# can also be found in the view layout files.
+odbcPreviewObject(channel, table="ANALYSIS.SMR01_PI", rowLimit=0)
+
+###############################################.
+## 1 - Look for data on an specific period of time ----
+###############################################.
+# In this example data is retrieved between the 1st and 2nd of April 16
+sql_example1 <- tbl_df(dbGetQuery(channel, statement=
+  "SELECT admission_date
+   FROM ANALYSIS.SMR01_PI 
+   WHERE admission_date between '1 April 2016' and '2 April 2016' ")) %>% 
```

Comment:
I haven't used to_date within SQL before to be honest. I see your point, and maybe there needs to be another section explaining more about dates in SQL and how use to_date, but for these examples I would favour not including it just because how easy is to read the dates and the statement.